### PR TITLE
indexer: remove diesel async

### PR DESCRIPTION
**Pull request by gegaowp**
## Description 

Diesel async was introduced to optimize indexer throughput but did not see significant improvement, also saw some issues on building like 
https://mysten-labs.slack.com/archives/C04FG4Q7YJ3/p1685265857488579

## Test Plan 

CI

local run to make sure indexer reader & writer can both work.

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

File: crates/sui-indexer/src/apis/indexer_api.rs [hunk 1]
```diff
@@ -449,19 +449,19 @@ where
...
+    async fn resolve_name_service_address(&self, name: String) -> RpcResult<Option<SuiAddress>> {
+        self.fullnode.resolve_name_service_address(name).await
     }
 
     async fn resolve_name_service_names(
         &self,
-        _address: SuiAddress,
-        _cursor: Option<ObjectID>,
-        _limit: Option<usize>,
+        address: SuiAddress,
```
Comment (wlmyng) [inline crates/sui-indexer/src/apis/indexer_api.rs:458]: am i hallucinating or was this already merged in another PR?

File: crates/sui-indexer/src/lib.rs [hunk 2]
```diff
@@ -6,27 +6,18 @@ use std::collections::HashMap;
...
-use std::time::SystemTime;
 
 use anyhow::{anyhow, Result};
 use axum::{extract::Extension, http::StatusCode, routing::get, Router};
-use backoff::future::retry;
 use backoff::ExponentialBackoff;
 use clap::Parser;
 use diesel::pg::PgConnection;
-use diesel::prelude::*;
+// use diesel::prelude::*;
```
Comment (wlmyng) [inline crates/sui-indexer/src/lib.rs:15]: nit: remove?

**Review by wlmyng — APPROVED** [review]

File: crates/sui-indexer/src/apis/indexer_api.rs (hunk 1 above)
Comment (gegaowp) [reply crates/sui-indexer/src/apis/indexer_api.rs]: ya, I just rebased, sry about the confusion lol

**Review by jnaulty — APPROVED** [review]
thanks @gegaowp  for resolving this sdk build issue!

**Pull request merged**